DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs

```diff
@@ -20,11 +20,136 @@ public class CourseServiceTests
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            var dto = sut.GetCourse(courseId);
+
+            //Than
+            Assert.AreEqual(courseDto, dto);
+            _courseRepositoryMock.Verify(x => x.GetCourse(1));
+        }
+        [Test]
+        public void GetCourseById_IntCourseId_ReturnCourseWithGroups_Topics_Materials_Tasks()
+        {
+            //Given
+            var courseDto = CourseData.GetCourseDto();
+            var courseId = CourseData.CourseId;
+            _courseRepositoryMock.Setup(x => x.GetCourse(courseId)).Returns(courseDto);
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            var dto = sut.GetFullCourseInfo(courseId);
+
+            //Than
+            Assert.AreEqual(courseDto, dto);
+            _courseRepositoryMock.Verify(x => x.GetCourse(courseId));
+        }
+
+        [Test]
+        public void GetCourses_WithoutParams_ReturnListCourses()
+        {
+            //Given
+            var courseDto = CourseData.GetListCourses();
+
+            _courseRepositoryMock.Setup(x => x.GetCourses()).Returns(courseDto);
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+
+            //When
+            var actualCourseDto = sut.GetCourses();
+
+            //Then
+            Assert.AreEqual(courseDto, actualCourseDto);
+            _courseRepositoryMock.Verify(x => x.GetCourses(), Times.Once);
+        }
+
+        [Test]
+        public void UpdateCourseInt_CourseId_Name_Descriptions_ReturnApdatedCourse()
+        {
+            //When
+            var courseId = CourseData.CourseId;
+            var courseDto = CourseData.GetCourseDto();
+            var updCourseDto = CourseData.GetUpdCourseDto();
+
+            _courseRepositoryMock.Setup(x => x.UpdateCourse(courseId, courseDto)).Returns(updCourseDto);
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+
+            //When
+            var actualCourseDto = sut.UpdateCourse(courseId, courseDto);
+
+            //Then
+            Assert.AreEqual(updCourseDto, actualCourseDto);
+            _courseRepositoryMock.Verify(x => x.UpdateCourse(courseId, courseDto), Times.Once);
```

Comment:
please add verification for methods:
![image](https://user-images.githubusercontent.com/18329725/127709633-d5beb876-6851-4d58-a4b8-8bea3ae7d8d3.png)

Reply:
+